### PR TITLE
Add new analytics service

### PR DESCRIPTION
**Pull request by WalshyDev**
## Description
As discussed in #developers, we want some better analytic insights into Slimefun (especially for the new storage layer work in progress).

This creates a new service and adds it to our current timings.

This can be debugged with `/sf debug slimefun_analytics`. Debugging will just show the JSON being sent and if the send was ignored.
Users can disable in the `config.yml` at `metrics.analytics`.

## Proposed changes
Creates a new analytics service

## Checklist
- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [ ] I have also tested the proposed changes in combination with various popular addons and can confirm my changes do not break them.
- [x] I have made sure that the proposed changes do not break compatibility across the supported Minecraft versions (1.16.* - 1.20.*).
- [x] I followed the existing code standards and didn't mess up the formatting.
- [x] I did my best to add documentation to any public classes or methods I added.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values
- [ ] I added sufficient Unit Tests to cover my code.

## Still to do

- [x] Whip up the analytic service and put it on GitHub
- [x] Test once the service is stood up

### CODE REVIEW

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            double totalTimings = Slimefun.getProfiler().getAndResetAverageNanosecondTimings();
+            double avgPerMachine = Slimefun.getProfiler().getAverageTimingsPerMachine();
+
+            if (totalTimings == 0 || avgPerMachine == 0) {
+                Debug.log(TestCase.ANALYTICS, "Ignoring analytics data for server_timings as no data was found"
+                    + " - total: " + totalTimings + ", avg: " + avgPerMachine);
+                // Ignore if no data
+                return;
+            }
+
```
Comment (J3fftw1) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java:61]: can we add build id.
So we know what timings go with which build

**Comment by WalshyDev** [comment]
Got a base site built out for this, just seeing if we can grab the slimefun.dev domain and then I can close this out

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/services/ThreadService.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        });
+    }
+
+    /**
+     * Invoke a new thread from the cached thread pool with the given name.
+     * This is a much better alternative to using
+     * {@link BukkitScheduler#runTaskAsynchronously(org.bukkit.plugin.Plugin, Runnable)}
+     * as this will show not only the plugin but a useful name.
+     * By default, Bukkit will use "Craft Scheduler Thread - <x> - <plugin>" which nice to show the plugin but
+     * it's impossible to track exactly what thread this is.
```
Comment (J3fftw1) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/services/ThreadService.java:41]:
```suggestion
     * By default, Bukkit will use "Craft Scheduler Thread - <x> - <plugin>" which is nice to show the plugin but
     * it's impossible to track exactly what thread that is.
```

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/services/ThreadService.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        });
+    }
+
+    /**
+     * Invoke a new scheduled thread from the cached thread pool with the given name.
+     * This is a much better alternative to using
+     * {@link BukkitScheduler#runTaskTimerAsynchronously(org.bukkit.plugin.Plugin, Runnable, long, long)}
+     * as this will show not only the plugin but a useful name.
+     * By default, Bukkit will use "Craft Scheduler Thread - <x> - <plugin>" which nice to show the plugin but
+     * it's impossible to track exactly what thread this is.
```
Comment (J3fftw1) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/services/ThreadService.java:62]:
```suggestion
     * By default, Bukkit will use "Craft Scheduler Thread - <x> - <plugin>" which is nice to show the plugin but
     * it's impossible to track exactly what thread that is.
```

**Review by J3fftw1 — CHANGES_REQUESTED** [review]

**Review by J3fftw1 — APPROVED** [review]
LGTM,  I really like this addition.
We can now see if our PRs actually improve code.

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                "AnalyticsService - Timings",
+                sendTimingsAnalytics(),
+                1,
+                1,
+                TimeUnit.MINUTES
+            );
+        }
+    }
+
+    // Timings we'll send the average every minute
```
Comment (Sfiguz7) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java:62]: Confused about what this means but may be cause I am tired

**Review by Sfiguz7 — CHANGES_REQUESTED** [review]
Typo and question because am ded

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java (hunk 4 above)
Comment (J3fftw1) [reply src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java:62]: That should be will iirc

Comment (WalshyDev) [reply src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java:62]: I wrote this while tired too 😅

Comment (WalshyDev) [reply src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java:62]:
```suggestion
    // We'll send some timing data every minute,
    // Today, we collect; the tick interval, the avg timing per tick and avg timing per machine
```

**Review by J3fftw1 — APPROVED** [review]

File: src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                sendTimingsAnalytics(),
+                1,
+                1,
+                TimeUnit.MINUTES
+            );
+        }
+    }
+
+    // We'll send some timing data every minute,
+    // Today, we collect; the tick interval, the avg timing per tick and avg timing per machine
```
Comment (Sfiguz7) [inline src/main/java/io/github/thebusybiscuit/slimefun4/core/services/AnalyticsService.java:63]:
```suggestion
    // We'll send some timing data every minute.
    // To date, we collect the tick interval, the avg timing per tick and avg timing per machine
```

**Review by Sfiguz7 — APPROVED** [review]

**Review by J3fftw1 — APPROVED** [review]

**Pull request merged**